### ISSUE TITLE
Stop/pause recognize command

### ISSUE THREAD

**Issue by fraschizzato** [sdkcarlos/artyom.js#5]
Needed the ability to stop/pause voice recognition. During speaking artyom recognizes it's own voice as command and I'd like to stop recognition while he speaks. command to have the ability to stop/resume only recognition. In my environment (artyom is configured to talk on app status change) if I stop artyom (fatality) he continues speaking (and thats ok) but with default language (english). It could be also usefull a configurable wakeup command, just to avoid false positive recognize. 
Thanks

**Comment by sdkcarlos** [comment]
Hey @fraschizzato i'm working on it but i see that the pause function (**use `.fatality` to stop**) may not work properly according to how you use it.

Please, download the dev-nodownload version of artyom and test what you want to achieve using the new methods. **I don't want to publish the 0.9.7 version yet**, so [if you want to test it quick download the dev-nodownload please here](https://github.com/sdkcarlos/artyom.js/blob/master/dev-no-download/artyom.js).

New methods : 

- **artyom.obey** : enable the command recognition if disabled.
- **artyom.dontObey** : disable the command recognition (not speechRecognition) dinamically.
- **artyom.initialize** (new property) : `obeyKeyword` property. If the command recognition is disabled with the dontObey method, it will be automatically enabled if you say the `obeyKeyword` while the command recognition is disabled (the suggested wakeUp command).

This will be published in the next version.

```javascript
// add the hello commands
artyom.addCommands({
    indexes: ["Hello"],
    action: function(i){
        console.log("Hey, i just said hello ... or maybe was the computer, snap!");
    }
});

// Start artyom how you always do ..
// But if you want the "wake up command"
// if artyom is not obeying, use the obeyKeyword property
artyom.initialize({
    lang:"en-GB",
    continuous:true,
    debug:true,
    listen:true,
    // if artyom is not obeying (paused) and the users says the obeyKeyword
    // it will continue with the command recognition (obey again)
    // if you say "start again" and artyom is not obeying, it will obey again.
    obeyKeyword: "start again"
});

// Talk something that may match with hello command
// Change the text with something of yours
artyom.say("Hello this message would normally match with a command.",{
    onStart:function(){
        // Don't obey any command
        artyom.dontObey();
    },
    onEnd: function(){
        setTimeout(function(){
            // Allow to process commands again
            artyom.obey();
        },2000);// wait 2 seconds
    }
});
```

However there are 2 behaviours that you need to know, if you're talking about "pause" that means
the recognition needs to be "active", therefore it will keep recognizing text till its end (unless you cancel or restart it). 
That will cause that the onResult event of the speechRecognition will keep the text previously recognized 
(even if artyomRecognition is paused) and it will processed again, triggering pitifully your command although it is paused.

Although what you want is difficult to achieve because the webkitSpeechRecognition cannot be paused (only methods : abort, start and stop), the processing can be
temporary paused while you use `artyom.say` using the onStart and onEnd callbacks.

If someone speaks after the onEnd callback, then it is most likely to keep saving the previous spoken text and it will fail.

Note : this works only in "normal" mode of artyom, with "quick" is not possible even without the timeout.

A simple solution would be to stop artyom when artyom says something and initialize it again (using timeout) because there's no other solution when you work in noisy environments (only with headphones would work :( ).

**Comment by fraschizzato** [comment]
Perfect, thanks

**Comment by sdkcarlos** [comment]
Function is now available since artyom 0.9.7 and the remote mode.
